### PR TITLE
Support custom replacement character

### PR DESCRIPTION
**Pull request by pseudomuto**
@csfrancis 

I've added two new methods to Rejectu that add support for specifying a replacement character to be used instead of the default `?`.

I'm sure there's a way I could do this with var args and use the two existing methods, but I thought I'd get your opinion on the idea first.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,14 +1,17 @@
 PATH
   remote: .
   specs:
-    rejectu (0.0.1)
+    rejectu (0.0.2)
 
 GEM
   remote: https://rubygems.org/
   specs:
+    power_assert (0.2.4)
```
Comment (csfrancis) [inline Gemfile.lock:9]: Do you need this?

**Comment by csfrancis** [comment]
I'm totally fine having an override for the replacement token, but like you hinted at, I'd rather do it with single method and var args. The approach seems reasonable to me though.

File: ext/rejectu/rejectu.c [hunk 2]
```diff
@@ -165,26 +166,38 @@ do_scrub(VALUE str)
...
   }
-  return do_scrub(str);
+  return do_scrub(str, rplToken);
 }
 
 static VALUE
-scrub_bang(VALUE self, VALUE str)
+scrub(VALUE self, VALUE str)
+{
+  return scrub_with_token(self, str, rb_str_new2("?"));
```
Comment (csfrancis) [inline ext/rejectu/rejectu.c]: I'd make the default `?` value a global for WEB SCALE.

Comment (pseudomuto) [inline ext/rejectu/rejectu.c]: :+1:

File: Gemfile.lock (hunk 1 above)
Comment (pseudomuto) [inline Gemfile.lock:9]: This is required by test-unit

**Comment by pseudomuto** [comment]
Updated to use var args and make the default token soooo WEB SCALE

File: ext/rejectu/rejectu.c [hunk 3]
```diff
@@ -7,6 +7,8 @@
 static VALUE mRejectu = Qnil;
 static VALUE idEncoding, idTo_s;
 
+static char *defaultToken = "?";
```
Comment (csfrancis) [inline ext/rejectu/rejectu.c]: What I meant was define `defaultToken` as a `VALUE` here and then initialize it in `Init_rejectu` with `rb_str_new2`.

**Comment by pseudomuto** [comment]
After adding `rb_global_variable(&defaultToken);` to mark the default as a global everything seems to be working correctly

**Comment by csfrancis** [comment]
:+1: LGTM.

**Pull request merged**